### PR TITLE
QgsServer Tests for JSON fields 

### PR DESCRIPTION
**Pull request by signedav**
Testing the implementation from #7869 with `GetFeatureInfo` requests over QGIS Server.

@elpaso Is it okay to have the tests in **test_qgsserver_wms_getfeatureinfo.py**? Would be no problem for me to put it into another file...

### CODE REVIEW

**Comment by pblottiere** [comment]
Hi @signedav,

> Is it okay to have the tests in test_qgsserver_wms_getfeatureinfo.py? Would be no problem for me to put it into another file...

I'm not @elpaso, but I think you did well :).

And thanks for taking the time to add server unit tests, very much appreciated :+1:!

**Comment by signedav** [comment]
Thanks @pblottiere 
I will fix the stuff Travis's complaining about today. Would it be possible that you review it then?

**Comment by signedav** [comment]
Thanks @pblottiere 
I will fix the stuff Travis's complaining about today. Would it be possible that you review it then?

**Comment by signedav** [comment]
Thanks @pblottiere 
I will fix the stuff Travis's complaining about today. Would it be possible that you review it then?

**Comment by signedav** [comment]
Btw. do you know what's the problem here https://travis-ci.org/qgis/QGIS/jobs/444553072#L1764 ?
Because this layer should be created here https://github.com/qgis/QGIS/blob/master/tests/testdata/provider/testdata_pg_json.sql so why it cannot connect to it?

**Comment by elpaso** [comment]
@signedav looks good, thanks!

**Comment by pblottiere** [comment]
> Btw. do you know what's the problem here https://travis-ci.org/qgis/QGIS/jobs/444553072#L1764 ?

I fetched your branch and everything is working well locally. I restarted the build, let's see what will happen...

File: tests/src/python/test_qgsserver_wms_getfeatureinfo.py [hunk 1]
```diff
@@ -1,4 +1,14 @@
 # -*- coding: utf-8 -*-
+from qgis.core import QgsProject
```
Comment (m-kuhn) [inline tests/src/python/test_qgsserver_wms_getfeatureinfo.py]: Can you move those imports back to where they were?

**Comment by m-kuhn** [comment]
@elpaso any idea why those tests are failing? It looks like the postgres layers are not available. Are there already any postgres based server tests in place?

**Comment by m-kuhn** [comment]
Notes from @elpaso 

> Check port number (I don't remember the DB setup on travis) and maybe 127.0.0.1 instead of localhost (I remember having issues with that)

Might also be an idea to try on an interactive travis debug session as @nyalldawson wrote recently [on the mailing list](http://osgeo-org.1560.x6.nabble.com/QGIS-Developer-Woohoo-Interactive-Travis-debugging-tp5381125.html)

File: tests/src/python/test_qgsserver_wms_getfeatureinfo.py (hunk 1 above)
Comment (signedav) [reply tests/src/python/test_qgsserver_wms_getfeatureinfo.py]: Python style tests where failing, so I did the formatting with autopep8 like proposed in the travis message. Don't you think it's incorrect style, when I change it back?

**Comment by pblottiere** [comment]
> any idea why those tests are failing? It looks like the postgres layers are not available. Are there already any postgres based server tests in place?

Yes, we already have numerous postgres based server tests in place. 

Moreover, I tested these news tests introduced by @signedav locally, and it works well (unless I made a mistake).

File: tests/src/python/test_qgsserver_wms_getfeatureinfo.py (hunk 1 above)
Comment (m-kuhn) [reply tests/src/python/test_qgsserver_wms_getfeatureinfo.py]: autopep8 takes a lot of configuration parameters (like line length and apparently placement of imports) which the style tests don't take into account.

The prepare-commit hook should run them properly, running autopep8 manually should never be required.

**Comment by m-kuhn** [comment]
@pblottiere can you point to those tests for comparison?

**Comment by pblottiere** [comment]
@m-kuhn 

For example the `test_project_values.qgz` project defines the layer `ls2d`:

```
dbname='qgis_test' port=5432 sslmode=disable key='tid' srid=4326 type=LineString table="qgis_test"."ls2d" (geom)
```

and this project/layer is used in `test_qgsserver_wms_getfeatureinfo.py`:

**Comment by signedav** [comment]
But in the datasources I still got user / password - I'll remove and push.

**Comment by m-kuhn** [comment]
:tada: thanks @pblottiere and @signedav !!

**Comment by signedav** [comment]
Thanks @pblottiere and @m-kuhn 

Just pushed the correct styling of python, but should be fine now.

**Comment by signedav** [comment]
:tada:

**Comment by pblottiere** [comment]
Thanks @signedav and @m-kuhn :D

**Pull request merged**